### PR TITLE
website: Terraform v1.3 upgrade guide

### PR DESCRIPTION
**Pull request by apparentlymart**
[Rendered Version](https://github.com/hashicorp/terraform/blob/upgrade-guide-v1.3/website/docs/language/upgrade-guides/index.mdx) (see also the Vercel-generated preview below, once it's complete.)

(Note: This PR is intentionally targeting the v1.3 branch directly, because upgrade guides are now release-branch-specific. The equivalent of this page on the `main` branch is a placeholder for the v1.4 upgrade guide which is not yet written.)

### CODE REVIEW

File: website/docs/language/upgrade-guides/index.mdx [hunk 1]
```diff
@@ -5,9 +5,120 @@ description: Upgrading to Terraform v1.3
...
+Terraform currently requires that all supported state storage backends be maintained in the Terraform codebase and compiled into Terraform CLI. Terraform therefore contains a mixture of backends maintained by the Terraform CLI team, backends maintained by other teams at HashiCorp, and backends maintained by third-party contributors.
+
+There are a number of backends that we have so far preserved on a best-effort basis despite them not having any active maintainers. Due to the overhead of continuing to support them, we deprecated the following unmaintained backends in Terraform v1.2.3:
+* `artifactory`
+* `etcd`
+* `etcdv3`
+* `manta`
+* `swift`
+
+All of these deprecated state storage backends are now removed in Terraform v1.3. If you are using any of these thing you will need to migrate to another state storage backend using Terraform v1.2 before you upgrade to Terraform v1.3.
```
[the remark this answers is not in the document]
Comment (apparentlymart) [reply website/docs/language/upgrade-guides/index.mdx:32]: Huh! Thanks for pointing this out; not sure what happened here!

File: website/docs/language/upgrade-guides/index.mdx [hunk 2]
```diff
@@ -5,9 +5,120 @@ description: Upgrading to Terraform v1.3
...
+
+The following sections describe some specific migration considerations for each removed backend.
+
+### Migrating from the `artifactory` backend
+
+From JFrog Artifactory 7.38.4 or later, Artifactory has support for the state storage protocol used by Terraform's "remote" backend, using a special repository type called a [Terraform Backend Repository](https://www.jfrog.com/confluence/display/JFROG/Terraform+Backend+Repository).
```
[the remark this answers is not in the document]
Comment (apparentlymart) [reply website/docs/language/upgrade-guides/index.mdx:38]: Hmm indeed I'm not sure why I didn't backtick this one... I was aiming to backtick all of the mentions of specific backend names in here. Thanks!

File: website/docs/language/upgrade-guides/index.mdx [hunk 3]
```diff
@@ -5,9 +5,120 @@ description: Upgrading to Terraform v1.3
 
 # Upgrading to Terraform v1.3
 
--> Do you need the upgrade guide for an earlier version of Terraform? Use the
-version selector in the navigation bar to select the version you are intending
-to upgrade to.
+-> Do you need the upgrade guide for a different version of Terraform? Use the version selector in the navigation bar to select the version you are intending to upgrade to.
```
Comment (laurapacilio) [inline website/docs/language/upgrade-guides/index.mdx:8]: We've been trying to steer away from saying "navigation bar" (instructions for the devdot migration) but I think we do need to call out the version selector here :) I'm also just wondering if we can make this note shorter -- you think this is okay or do you think we need to be more explicit?

Another thing that Judith and I thought of today that I wanted to flag for you --> Will anyone be jumping up more than one version? If so, can they come straight to this one or are there versions in between that they need to go through the migration process first. Basically - anything folks should know who are still on like v1.1 or earlier? 

```suggestion
-> **Note:** Use the version selector to view the upgrade guides for older Terraform versions. 
```

Comment (laurapacilio) [reply website/docs/language/upgrade-guides/index.mdx]: It's okay! We can fix forward! I'm not that concerned about this note, although maybe we should go back like ONE more to v1.2? 

I'm okay with not including anything, I more poked at it to just make sure we thought of it/see what the response was. I guess my ask is --> if it ever becomes a time when we're like "okay so you really DO need to upgrade to version X before this one if you haven't already", can we just say that to users? I'll trust your judgment about when/if that is ever needed. But I like less text, so if it's not helpful/relevant to include right now, let's not :)

Comment (apparentlymart) [reply website/docs/language/upgrade-guides/index.mdx]: I think a specific goal of the 1.x series is that it should always be possible and easy for most users to just jump to the latest 1.x without worrying about intermediate steps. Of course these upgrade guides do capture some situations where a small number of users may need to take a specific action when upgrading, but we're aiming to keep these small enough in scope that relatively few people will actually _need_ the upgrade guides.

In the worst case a subset of the already-smaller number of folks who needed the upgrade guides will not find the upgrade guides and will ask a question in a community forum space. Due to this being a subset of an already-small subset of our users, I think we've done okay with just helping those users find the upgrade guides one-on-one when it arises, and so I think we currently have a good tradeoff for this.

It is possible that one day there is a 2.x release which could once again act as a "stopping point" during upgrades if it has changed significantly enough to require explicit upgrade steps. But at this time we have no plans for any such changes and in fact our current goal is to do exactly the opposite: to continue the 1.x series indefinitely and use backward-compatible or explicit-opt-in techniques to introduce new features and changes. In the event that we change our minds and _do_ find the need or benefit for a 2.x break, I think we can cross that bridge when we get there. :grinning:

**Pull request merged**